pkg/services/automation/generator.go

```diff
@@ -0,0 +1,494 @@
+	"github.com/fluxcd/source-controller/pkg/sourceignore"
+	"github.com/weaveworks/weave-gitops/pkg/flux"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/logger"
+	"github.com/weaveworks/weave-gitops/pkg/models"
+
+	wego "github.com/weaveworks/weave-gitops/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	MaxKubernetesResourceNameLength = 63
+
+	WeGOAppIdentifierLabelKey = "wego.weave.works/app-identifier"
+)
+
+type ConfigMode string
+
+type SourceType string
+
+type ResourceKind string
+
+type ResourceRef struct {
+	Kind           ResourceKind
+	Name           string
+	RepositoryPath string
+}
+
+const (
+	ConfigModeClusterOnly  ConfigMode = "clusterOnly"
+	ConfigModeUserRepo     ConfigMode = "userRepo"
+	ConfigModeExternalRepo ConfigMode = "externalRepo"
+
+	ResourceKindApplication    ResourceKind = "Application"
+	ResourceKindSecret         ResourceKind = "Secret"
+	ResourceKindGitRepository  ResourceKind = "GitRepository"
+	ResourceKindHelmRepository ResourceKind = "HelmRepository"
+	ResourceKindKustomization  ResourceKind = "Kustomization"
+	ResourceKindHelmRelease    ResourceKind = "HelmRelease"
+)
+
+type AutomationGenerator interface {
+	GenerateAutomation(ctx context.Context, app models.Application, clusterName string) ([]AutomationManifest, error)
+}
+
+type AutomationGen struct {
+	GitProvider gitproviders.GitProvider
+	Flux        flux.Flux
+	Logger      logger.Logger
+}
+
+var _ AutomationGenerator = &AutomationGen{}
+
+type AutomationManifest struct {
+	Path    string
+	Content []byte
+}
+
+func NewAutomationGenerator(gp gitproviders.GitProvider, flux flux.Flux, logger logger.Logger) AutomationGenerator {
+	return &AutomationGen{
+		GitProvider: gp,
+		Flux:        flux,
+		Logger:      logger,
+	}
+}
+
+func (a *AutomationGen) getAppSecretRef(ctx context.Context, app models.Application, clusterName string) (GeneratedSecretName, error) {
+	if app.SourceType != models.SourceTypeHelm {
+		return a.getSecretRef(ctx, app, app.GitSourceURL, clusterName)
+	}
+
+	return "", nil
+}
+
+func (a *AutomationGen) getSecretRef(ctx context.Context, app models.Application, url gitproviders.RepoURL, clusterName string) (GeneratedSecretName, error) {
+	var secretRef GeneratedSecretName
+
+	visibility, visibilityErr := a.GitProvider.GetRepoVisibility(ctx, url)
+	if visibilityErr != nil {
+		return "", visibilityErr
+	}
+
+	if *visibility != gitprovider.RepositoryVisibilityPublic {
+		secretRef = CreateRepoSecretName(clusterName, url)
+	}
+
+	return secretRef, nil
+}
+
+func (a *AutomationGen) generateAppAutomation(ctx context.Context, app models.Application, clusterName string) ([]AutomationManifest, error) {
+	a.Logger.Generatef("Generating GitOps automation manifests")
```

Comment:
This log applies to the `generateApplicationGoat` call

Reply:
Neglected to move comments after refactoring